### PR TITLE
Disable closing of the overlay by clicking on notification content

### PR DESCRIPTION
**Pull request by yuriy-fix**
Fixes https://github.com/vaadin/vaadin-notification/issues/56

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-overlay/70)

### CODE REVIEW

**Comment by platosha** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[demo/index.html, line 209 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70#-L82oebAc37OB2LFGMwk:-L82oebAc37OB2LFGMwl:bl73uka) ([raw file](https://github.com/vaadin/vaadin-overlay/blob/88c3cd79084436f158b915160919f8f996ea0a80/demo/index.html#L209)):*
> ```HTML
> 
> 
>     <h3>Overlay opened along with notification</h3>
> ```

Titles Should Follow Title Case

---

*[test/vaadin-overlay.html, line 68 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70#-L82pOmT6_SEzNMcwGqj:-L82pOmT6_SEzNMcwGqk:b-7qxk8g) ([raw file](https://github.com/vaadin/vaadin-overlay/blob/88c3cd79084436f158b915160919f8f996ea0a80/test/vaadin-overlay.html#L68)):*
> ```HTML
>     <template>
>       <div>
>         <div id="parent">
> ```

Unnecessary wrapper divs.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70)*

**Comment by yuriy-fix** [comment]
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[demo/index.html, line 209 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70#-L82oebAc37OB2LFGMwk:-L82vHyIEsBSKZ3VmEvM:b-896fix) ([raw file](https://github.com/vaadin/vaadin-overlay/blob/88c3cd79084436f158b915160919f8f996ea0a80/demo/index.html#L209)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Titles Should Follow Title Case
</blockquote></details>

Done.

---

*[test/vaadin-overlay.html, line 68 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70#-L82pOmT6_SEzNMcwGqj:-L82vIRpHk5GttyWwK9q:b-896fix) ([raw file](https://github.com/vaadin/vaadin-overlay/blob/88c3cd79084436f158b915160919f8f996ea0a80/test/vaadin-overlay.html#L68)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Unnecessary wrapper divs.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70)*

File: src/vaadin-overlay.html [hunk 1]
```diff
@@ -322,7 +322,8 @@
           this._mouseUpInside = false;
           return;
         }
-        if (!this._last) {
+
+        if (!this._last || event.composedPath().filter(el => el.localName === 'vaadin-notification-card')[0]) {
```
Comment (limonte) [inline src/vaadin-overlay.html:326]: bad idea, `vaadin-overlay` should not know anything about components which are using it.

**Review by limonte — CHANGES_REQUESTED** [review]

**Comment by yuriy-fix** [comment]
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/vaadin-overlay.html, line 326 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70#-L82nOaBRoGc9Sx522DP-r2-326:-L835_-m6hfy1QVTA-Pr:b-xxbljh) ([raw file](https://github.com/vaadin/vaadin-overlay/blob/12a59c8de4d4dbd955a610f5955f6847f3ee38e0/src/vaadin-overlay.html#L326)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

bad idea, `vaadin-overlay` should not know anything about components which are using it.
</blockquote></details>

It's a specific case, as trying to solve it from the other side (`vaadin-notification` which is not using `vaadin-overlay`) will require to interfere with capture mode inside it, listening on the document etc.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70)*

**Comment by limonte** [comment]
Review status: 2 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/vaadin-overlay.html, line 326 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70#-L82nOaBRoGc9Sx522DP-r2-326:-L87NxgLkyaySVGVbUUq:bqjcmql) ([raw file](https://github.com/vaadin/vaadin-overlay/blob/12a59c8de4d4dbd955a610f5955f6847f3ee38e0/src/vaadin-overlay.html#L326)):*
<details><summary><i>Previously, YuriyVaadin (Yuriy Yevstihnyeyev) wrote…</i></summary><blockquote>

It's a specific case, as trying to solve it from the other side (`vaadin-notification` which is not using `vaadin-overlay`) will require to interfere with capture mode inside it, listening on the document etc.
</blockquote></details>

Unfortunately, I can't suggest any better approach here, but this one is bad. The fact that we don't have a good solution doesn't mean that we should merge the bad one. This is a blocker.

`vaadin-overlay` absolutely should not depend on component names of the external world. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70)*

**Comment by yuriy-fix** [comment]
@platosha , @tomivirkki , @limonte , What should I do with this PR? Should I close it?

---

Review status: 2 of 4 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/vaadin/vaadin-overlay/70#-:-L8HBc9k7dk9Y3a_dGOA:b2kwa98)*

**Comment by web-padawan** [comment]
I do not like the solution but seems that we should keep this as a workaround for platform. So LGTM by me.

---

Reviewed 2 of 4 files at r1, 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-overlay/70#-:-L8HG0DRw5gmzzSce8Ue:b-gbmgii)*

**Comment by yuriy-fix** [comment]
Closed in purpose of future investigation.

**Pull request closed without merging**